src/ui/ui_transaction_nbgl.c

```diff
@@ -0,0 +1,260 @@
+
+    // Prepare formatter stack.
+    formatter_stack[0] = get_formatter(&G_context.tx_info, true);  // SET FORMATTERS STACK
+
+    while (continue_loop) {  // Execute loop until last tx formatter is reached.
+        explicit_bzero(G.ui.detail_caption, sizeof(G.ui.detail_caption));
+        explicit_bzero(G.ui.detail_value, sizeof(G.ui.detail_value));
+        explicit_bzero(op_caption, sizeof(op_caption));
+        previous_idx = formatter_index;
+        previous_data = G.ui.current_data_index;
+        // Call formatter function.
+        formatter_stack[formatter_index](&G_context.tx_info);
+        PRINTF("Page %d - Item : %s - Value : %s\n",
+               nbPages,
+               G.ui.detail_caption,
+               G.ui.detail_value);
+        // Compute number of lines filled by tag item string.
+        fieldLen = strlen(G.ui.detail_caption);
+        tagItemLineNb = fieldLen / TAG_VAL_LST_ITEM_MAX_CHAR_PER_LINE;
+        tagItemLineNb += (fieldLen % TAG_VAL_LST_ITEM_MAX_CHAR_PER_LINE != 0) ? 1 : 0;
+        // Compute number of lines filled by tag value string.
+        fieldLen = strlen(G.ui.detail_value);
+        tagValueLineNb = fieldLen / TAG_VAL_LST_VAL_MAX_CHAR_PER_LINE;
+        tagValueLineNb += (fieldLen % TAG_VAL_LST_VAL_MAX_CHAR_PER_LINE != 0) ? 1 : 0;
+        // Add number of screen lines occupied by tag pair to total lines occupied in page.
+        tagLineNb = tagValueLineNb + tagItemLineNb;
+        pageLineNb += tagLineNb;
+        // If there are multiple operations and a new operation is reached, create a
+        // special page with only one caption/value pair to display operation number.
+        if (G.ui.current_data_index > previous_data &&
+            G_context.tx_info.tx_details.operations_count > 1) {
+            nbPages++;
+            pagesInfos[nbPages].pagePairNb = 1;
+            pagesInfos[nbPages].formatter_index = previous_idx;
+            pagesInfos[nbPages].data_idx = previous_data;
+            pagesInfos[nbPages].centered_info = true;
+            nbPages++;
+            pageLineNb = 0;
+            pagesInfos[nbPages].pagePairNb = 0;
+            pagesInfos[nbPages].formatter_index = formatter_index + 1;
+            pagesInfos[nbPages].data_idx = G.ui.current_data_index;
+        }
+        // Else if number of lines occupied on page > allowed max number of lines per page,
+        // go to next page.
+        else if (pageLineNb > TAG_VAL_LST_MAX_LINES_PER_PAGE) {
+            nbPages++;
+            pageLineNb = tagLineNb;
+            pagesInfos[nbPages].pagePairNb = 1;
+            pagesInfos[nbPages].formatter_index = formatter_index;
+            pagesInfos[nbPages].data_idx = G.ui.current_data_index;
+        } else
+        // Otherwise save number of pairs on current page
+        {
+            pagesInfos[nbPages].pagePairNb++;
+        }
+        formatter_index++;
+        continue_loop = (formatter_stack[formatter_index] != NULL) ? true : false;
+    }
+
+    nbPages++;
```

Comment:
it seems that `nbPages `is never checked (should not exceed `pagesInfos` size)

Reply:
Fixed.